### ISSUE TITLE
clang binary settings not allowing whitespace in path

### ISSUE THREAD

**Issue by nolange** [niosus/EasyClangComplete#458]
Sorry for not adhering to the issue template, but I will make up by providing the fix.
The issue is that on windows you have a path like `"clang_binary" : "C:\\Program Files\\LLVM\\bin\\clang++.exe"`

the ECC plugin will try reading the version by executing `"C:\\Program" "Files\\LLVM\\bin\\clang++.exe" "-v"`. Doesnt work obviously.

Fix would be to change the following lines in tools.py:705

        check_version_cmd = [ clang_binary, "-v" ]
        log.info("Getting version from command: `%s`", ' '.join(check_version_cmd))

**Comment by nolange** [comment]
Probably the same issue with `cmake_binary` path, but that seems more involved.
(Could not get this plug to work on windows yet, I know this is on open issue)

**Comment by niosus** [comment]
I do want to see the log for that.
We do run the tests on Windows and it is expected that the system works out of the box on Windows if you have visual c++ installed. The builds seem to work fine and get the version properly: https://ci.appveyor.com/project/niosus/easyclangcomplete

I don't understand why the executing `"C:\\Program" "Files\\LLVM\\bin\\clang++.exe" "-v"` does not work.

**Comment by nolange** [comment]
Dont know what "Visual C++" means, the IDE, the compiler, which version?

I am using the Build tools (No IDE), official Clang 6.0.0 and integration in the MSBuild system (as the 2017 Build tools aren't currrently automatically supported).

eg, with CMake I have to configure like that: `"C:\Program Files\CMake\bin\cmake.exe" -G"Visual Studio 15 2017" -T"LLVM-vs2017" F:\poeinfo`

https://aka.ms/BuildTools
http://releases.llvm.org/download.html
https://github.com/arves100/llvm-vs2017-integration

> I don't understand why the executing "C:\\Program" "Files\\LLVM\\bin\\clang++.exe" "-v" does not work.
It tries to execute a file "C:\\Program" with the rest as arguments. This would be an issue in Linux aswell - try symlinking clang `ln -s /usr/bin/clang++ /home/you/some path/clang++` and set the path to the variable clang_binary

You now got 2 logs, one without adjusting settings, one with the paths below, and one with the paths und the suggested fix (fails with cmakes path)

```
{
	"clang_binary" : "C:\\Program Files\\LLVM\\bin\\clang++.exe",
	"cmake_binary" : "C:\\Program Files\\CMake\\bin\\cmake.exe",
	"libclang_path": "C:\\Program Files\\LLVM\\bin\\libclang.dll"
}
```

<details>
<summary>Default log (no settings): </summary>
  
```
DPI scale: 1
startup, version: 3143 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.248689
startup time: 0.311689
first paint time: 0.311689
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin BracketHighlighter.bh_core
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_popup
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin BracketHighlighter.support
reloading plugin Clang Format.clang_format
reloading plugin CMake.AutoComplete
reloading plugin DocBlockr.jsdocs
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin Pandoc.Pandoc
reloading plugin VimModelines.vimmodelines
reloading plugin EasyClangComplete.EasyClangComplete
plugins loaded
Loaded VimModelines
[ECC:DEBUG]:[settings_storage.py]:[need_reparse]:[MainThread]: settings complete
[ECC:DEBUG]:[EasyClangComplete.py]:[on_activated_async]:[MainThread]: on_activated_async view id 26
[ECC:DEBUG]:[settings_manager.py]:[settings_for_view]:[MainThread]: no settings for view 26. Reinitializing.
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Overriding settings by project ones if needed:
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Valid prefixes: ['ecc_', 'easy_clang_complete_']
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Reading settings...
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Settings sucessfully read...
[ECC:INFO]:[tools.py]:[get_clang_version_str]:[MainThread]: Getting version from command: `clang++ -v`
[ECC:DEBUG]:[tools.py]:[run_command]:[MainThread]: command finished with code: 1
[ECC:DEBUG]:[tools.py]:[run_command]:[MainThread]: command output: 
'clang++' is not recognized as an internal or external command,
operable program or batch file.

Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 210, in on_api_ready
    m.plugin_loaded()
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\EasyClangComplete.py", line 77, in plugin_loaded
    handle_plugin_loaded_function()
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\EasyClangComplete.py", line 214, in on_plugin_loaded
    self.on_activated_async(sublime.active_window().active_view())
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\EasyClangComplete.py", line 265, in on_activated_async
    settings = EasyClangComplete.settings_manager.settings_for_view(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_manager.py", line 55, in settings_for_view
    self.__init_for_view(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_manager.py", line 103, in __init_for_view
    self.__settings_dict[view_id].update_from_view(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_storage.py", line 123, in update_from_view
    self.__update_wildcard_values(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_storage.py", line 311, in __update_wildcard_values
    version_str = Tools.get_clang_version_str(self.clang_binary)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\tools.py", line 712, in get_clang_version_str
    return cls._get_regular_clang_version_str(output_text)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\tools.py", line 723, in _get_regular_clang_version_str
    raise RuntimeError(" Couldn't find clang version in clang version "
RuntimeError:  Couldn't find clang version in clang version output.
Package Control: Skipping automatic upgrade, last run at 2018-04-20 12:55:15, next run at 2018-04-20 13:55:15 or after
[ECC:DEBUG]:[view_config.py]:[trigger_info]:[Thread-8]: Config is not ready yet. No info tooltip shown.
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-9]: view file_name is None
```

</details>

<details>
<summary>Settings adjusted log: </summary>
  
```
DPI scale: 1
startup, version: 3143 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.250677
startup time: 0.313677
first paint time: 0.313677
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin BracketHighlighter.bh_core
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_popup
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin BracketHighlighter.support
reloading plugin Clang Format.clang_format
reloading plugin CMake.AutoComplete
reloading plugin DocBlockr.jsdocs
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin Pandoc.Pandoc
reloading plugin VimModelines.vimmodelines
reloading plugin EasyClangComplete.EasyClangComplete
plugins loaded
[ECC:DEBUG]:[settings_storage.py]:[need_reparse]:[MainThread]: settings complete
[ECC:DEBUG]:[EasyClangComplete.py]:[on_activated_async]:[MainThread]: on_activated_async view id 26
[ECC:DEBUG]:[settings_manager.py]:[settings_for_view]:[MainThread]: no settings for view 26. Reinitializing.
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Overriding settings by project ones if needed:
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Valid prefixes: ['ecc_', 'easy_clang_complete_']
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Reading settings...
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Settings sucessfully read...
[ECC:INFO]:[tools.py]:[get_clang_version_str]:[MainThread]: Getting version from command: `C:\Program Files\LLVM\bin\clang++.exe -v`
[ECC:DEBUG]:[tools.py]:[run_command]:[MainThread]: command finished with code: 1
[ECC:DEBUG]:[tools.py]:[run_command]:[MainThread]: command output: 
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.

Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 210, in on_api_ready
    m.plugin_loaded()
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\EasyClangComplete.py", line 77, in plugin_loaded
    handle_plugin_loaded_function()
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\EasyClangComplete.py", line 214, in on_plugin_loaded
    self.on_activated_async(sublime.active_window().active_view())
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\EasyClangComplete.py", line 265, in on_activated_async
    settings = EasyClangComplete.settings_manager.settings_for_view(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_manager.py", line 55, in settings_for_view
    self.__init_for_view(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_manager.py", line 103, in __init_for_view
    self.__settings_dict[view_id].update_from_view(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_storage.py", line 123, in update_from_view
    self.__update_wildcard_values(view)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\settings\settings_storage.py", line 311, in __update_wildcard_values
    version_str = Tools.get_clang_version_str(self.clang_binary)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\tools.py", line 712, in get_clang_version_str
    return cls._get_regular_clang_version_str(output_text)
  File "C:\Users\vienor05\AppData\Roaming\Sublime Text 3\Packages\EasyClangComplete\plugin\tools.py", line 723, in _get_regular_clang_version_str
    raise RuntimeError(" Couldn't find clang version in clang version "
RuntimeError:  Couldn't find clang version in clang version output.
Loaded VimModelines
Package Control: Skipping automatic upgrade, last run at 2018-04-20 12:55:15, next run at 2018-04-20 13:55:15 or after
[ECC:DEBUG]:[view_config.py]:[trigger_info]:[Thread-8]: Config is not ready yet. No info tooltip shown.
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-9]: view file_name is None

```

</details>

<details>
<summary>Settings adjusted + my fix log: </summary>
  
```
DPI scale: 1
startup, version: 3143 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/vienor05/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.246714
startup time: 0.309713
first paint time: 0.309713
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin BracketHighlighter.bh_core
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_popup
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin BracketHighlighter.support
reloading plugin Clang Format.clang_format
reloading plugin CMake.AutoComplete
reloading plugin DocBlockr.jsdocs
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin Pandoc.Pandoc
reloading plugin VimModelines.vimmodelines
reloading plugin EasyClangComplete.EasyClangComplete
plugins loaded
[ECC:DEBUG]:[settings_storage.py]:[need_reparse]:[MainThread]: settings complete
[ECC:DEBUG]:[EasyClangComplete.py]:[on_activated_async]:[MainThread]: on_activated_async view id 26
[ECC:DEBUG]:[settings_manager.py]:[settings_for_view]:[MainThread]: no settings for view 26. Reinitializing.
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Overriding settings by project ones if needed:
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Valid prefixes: ['ecc_', 'easy_clang_complete_']
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Reading settings...
[ECC:DEBUG]:[settings_storage.py]:[__load_vars_from_settings]:[MainThread]: Settings sucessfully read...
[ECC:INFO]:[tools.py]:[get_clang_version_str]:[MainThread]: Getting version from command: `C:\Program Files\LLVM\bin\clang++.exe -v`
[ECC:DEBUG]:[tools.py]:[run_command]:[MainThread]: running command: 
"C:\Program Files\LLVM\bin\clang++.exe" -v
[ECC:DEBUG]:[settings_storage.py]:[__populate_common_flags]:[MainThread]: populating common_flags with current variables.
[ECC:DEBUG]:[settings_storage.py]:[__replace_wildcard_if_needed]:[MainThread]: populated '-I$project_base_path/src' to '-I/src'
[ECC:DEBUG]:[settings_storage.py]:[__replace_wildcard_if_needed]:[MainThread]: populated '-I/usr/lib/clang/$clang_version/include' to '-I/usr/lib/clang/6.0.0/include'
[ECC:DEBUG]:[settings_manager.py]:[__init_for_view]:[MainThread]: settings initialized for view: 26
Loaded VimModelines
[ECC:DEBUG]:[view_config.py]:[load_for_view]:[Thread-5]: generate new config for path: 26
[ECC:INFO]:[view_config.py]:[__init_completer]:[Thread-5]: init completer based on libclang
[ECC:DEBUG]:[lib_complete.py]:[__init__]:[Thread-5]: using bundled cindex: EasyClangComplete.plugin.clang.cindex50
[ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-5]: platform: ('64bit', 'WindowsPE')
[ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-5]: python version: 3.3.6
[ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-5]: we are on 'Windows'
[ECC:DEBUG]:[utils.py]:[find_libclang_dir]:[Thread-5]: user provided libclang_path: C:\Program Files\LLVM\bin\libclang.dll
[ECC:INFO]:[utils.py]:[find_libclang_dir]:[Thread-5]: using user-provided libclang: 'C:\Program Files\LLVM\bin\libclang.dll'
[ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-5]: [cmake]:[get]: for file F:\poeinfo\poeinfo.c
[ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-5]: [cmake]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-5]: searching 'CMakeLists.txt' from 'F:\poeinfo' to 'C:\'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-5]: found 'CMakeLists.txt' file: F:\poeinfo\CMakeLists.txt
[ECC:DEBUG]:[tools.py]:[contains]:[Thread-5]: found needed line: 'project(poeinfo VERSION 0.1)
'
[ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-5]: [cmake]:[current]: 'F:\poeinfo\CMakeLists.txt'
[ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-5]: [cmake]:[generate new db]
[ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-5]: Folder c:\users\vienor05\appdata\local\temp\EasyClangComplete\cmake_builds\761c88aa64d4e95cc5da4f3848f965da exists.
[ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-5]: prefix paths: []
[ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-5]: merged paths: 
[ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-5]: CMAKE_PREFIX_PATH: 
[ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-5]:  running command: C:\Program Files\CMake\bin\cmake.exe -DCMAKE_EXPORT_COMPILE_COMMANDS=ON  "F:\poeinfo"
[ECC:DEBUG]:[tools.py]:[run_command]:[Thread-5]: command finished with code: 1
[ECC:DEBUG]:[tools.py]:[run_command]:[Thread-5]: command output: 
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.

[ECC:DEBUG]:[cmake_file.py]:[__compile_cmake]:[Thread-5]: cmake produced output: 
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.

[ECC:ERROR]:[cmake_file.py]:[__compile_cmake]:[Thread-5]: cmake has finished, but no compilation database.
[ECC:DEBUG]:[compilation_db.py]:[get_flags]:[Thread-5]: [db]:[get]: for file f:\poeinfo\poeinfo.c
[ECC:DEBUG]:[compilation_db.py]:[get_flags]:[Thread-5]: [db]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-5]: searching 'compile_commands.json' from 'F:\poeinfo' to 'C:\'
[ECC:DEBUG]:[compilation_db.py]:[get_flags]:[Thread-5]: [db]:[current]: 'None'
[ECC:DEBUG]:[compilation_db.py]:[get_flags]:[Thread-5]: [db]:[load new]
[ECC:DEBUG]:[compilation_db.py]:[get_flags]:[Thread-5]: [db]:[no new]: return None
[ECC:DEBUG]:[CppProperties.py]:[get_flags]:[Thread-5]: [CppProperties]:[get]: for file F:\poeinfo\poeinfo.c
[ECC:DEBUG]:[CppProperties.py]:[get_flags]:[Thread-5]: [CppProperties]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-5]: searching 'CppProperties.json' from 'F:\poeinfo' to 'C:\'
[ECC:DEBUG]:[CppProperties.py]:[get_flags]:[Thread-5]: [CppProperties]:[current]: 'None'
[ECC:DEBUG]:[c_cpp_properties.py]:[get_flags]:[Thread-5]: [c_cpp_properties]:[get]: for file F:\poeinfo\poeinfo.c
[ECC:DEBUG]:[c_cpp_properties.py]:[get_flags]:[Thread-5]: [c_cpp_properties]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-5]: searching 'c_cpp_properties.json' from 'F:\poeinfo' to 'C:\'
[ECC:DEBUG]:[c_cpp_properties.py]:[get_flags]:[Thread-5]: [c_cpp_properties]:[current]: 'None'
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-5]: [clang_complete_file]:[get]: for file F:\poeinfo\poeinfo.c
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-5]: [clang_complete_file]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-5]: searching '.clang_complete' from 'F:\poeinfo' to 'C:\'
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-5]: [clang_complete_file]:[current]: 'None'
[ECC:DEBUG]:[lib_complete.py]:[update]:[Thread-5]: view is 26
[ECC:DEBUG]:[lib_complete.py]:[update]:[Thread-5]: translation unit does not exist. Creating.
[ECC:DEBUG]:[lib_complete.py]:[parse_tu]:[Thread-5]: clang flags are: ['-c', '-fsyntax-only', '-x', 'c', '-std=c11', '-I\\usr\\include', '-I\\src', '-I\\usr\\lib\\clang\\6.0.0\\include', '-IF:\\poeinfo', '-IF:\\']
[ECC:DEBUG]:[lib_complete.py]:[parse_tu]:[Thread-5]: compilation started for view id: 26
[ECC:DEBUG]:[lib_complete.py]:[parse_tu]:[Thread-5]: compilation done in 0.1250150203704834 seconds
[ECC:DEBUG]:[lib_complete.py]:[update]:[Thread-5]: reparsing translation_unit for view 26
[ECC:DEBUG]:[singleton_thread_pool.py]:[new_job]:[MainThread]: Canceled job: 'job: 'info''
[ECC:DEBUG]:[lib_complete.py]:[update]:[Thread-5]: reparsed in 0.03125190734863281 seconds
Package Control: Skipping automatic upgrade, last run at 2018-04-20 12:55:15, next run at 2018-04-20 13:55:15 or after
[ECC:DEBUG]:[popup_error_vis.py]:[generate]:[Thread-5]: generating error regions for view 26
[ECC:DEBUG]:[popup_error_vis.py]:[generate]:[Thread-5]: 1 error regions ready
[ECC:DEBUG]:[popup_error_vis.py]:[show_errors]:[Thread-5]: showing error regions: []
[ECC:DEBUG]:[EasyClangComplete.py]:[config_updated]:[Thread-5]: updated config: <EasyClangComplete.plugin.view_config.ViewConfig object at 0x000002A9B3E09F28>
[ECC:DEBUG]:[view_config.py]:[get_from_cache]:[Thread-5]: config exists for view: 26
[ECC:DEBUG]:[view_config.py]:[get_from_cache]:[Thread-5]: config: <EasyClangComplete.plugin.view_config.ViewConfig object at 0x000002A9B3E09F28>
[ECC:DEBUG]:[view_config.py]:[get_from_cache]:[Thread-5]: config exists for view: 26
[ECC:DEBUG]:[view_config.py]:[get_from_cache]:[Thread-5]: config: <EasyClangComplete.plugin.view_config.ViewConfig object at 0x000002A9B3E09F28>
[ECC:DEBUG]:[tools.py]:[is_valid_view]:[Dummy-10]: view file_name is None

```

</details>

**Comment by niosus** [comment]
Ah, ok. I missed the fact that it breaks up the Program Files folder in two. We need to sanitize the input from these folders. Right.

**Comment by niosus** [comment]
Should be closed with #466
